mk/xamarin.mk

```diff
@@ -1,6 +1,6 @@
 ifdef ENABLE_XAMARIN
-NEEDED_MACCORE_VERSION := 62c4393c94ed274acbc1edce259a3c22a461f922
-NEEDED_MACCORE_BRANCH := master
+NEEDED_MACCORE_VERSION := 5a97b45faf02315eb266d3670fd1492e34228f0f
+NEEDED_MACCORE_BRANCH := 2017-12
```

Comment:
this will need to be merged first (once it's all green) on maccore

Reply:
Do you want to go through a PR as well for `xamarin/maccore`?

Reply:
yes, but there's no build on maccore so it can be fast - mostly just reviewing the changes. The original ones were minor, only mono-extensions bump, but could not be merged ahead of time since it broke with 2017-10